cmd/README.md

```diff
@@ -0,0 +1,608 @@
+# Precompile Generation Tutorial
+
+We can now generate a stateful precompile with the Precompile gen tool!
+
+### Assumption of Knowledge
+Before starting this tutorial it would be helpful if you had some context on the EVM, precompiles, and stateful precompiles. 
+Here are some resources to get started put together. 
+
+- [EVM Handbook](https://noxx3xxon.notion.site/noxx3xxon/The-EVM-Handbook-bb38e175cc404111a391907c4975426d)
+
+- [Precompiles in Solidity](https://medium.com/@rbkhmrcr/precompiles-solidity-e5d29bd428c4)
+
+- [Customizing the EVM with Stateful Precompiles](https://medium.com/avalancheavax/customizing-the-evm-with-stateful-precompiles-f44a34f39efd)
+
+ 
+## Tutorial
+
+Let's start by creating the Solidity interface that we want to implement. We can put this  in `./contract-examples/contracts`
+
+```
+// (c) 2022-2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+// SPDX-License-Identifier: MIT
+
+pragma solidity >=0.8.0;
+
+interface IHelloWorld {
+  function sayHello() external returns (string calldata);
+
+  function setGreeting(string calldata recipient) external;
+}
+```
+
+Now we have an interface that our precompile can implement!
+Let's create an [abi](https://docs.soliditylang.org/en/v0.8.13/abi-spec.html#:~:text=Contract%20ABI%20Specification-,Basic%20Design,as%20described%20in%20this%20specification.) of our solidity code.
+
+In the same `./contract-examples/contracts` directory, let's run
+
+```
+solcjs --abi IHelloWorld.sol
+```
```

Comment:
We should add a section here that explains how to install the solidity compiler: https://docs.soliditylang.org/en/latest/installing-solidity.html and say that you should install it before moving on.

Can we switch to using the more complete `solc` instead of `solcjs` in this tutorial? They mention in the docs at that same link that their later documentation on using the compiler assumes that the full `solc` has been installed.

Reply:
yup